### PR TITLE
APP 153 [C] Fix notification demo

### PR DESCRIPTION
**Pull request by Dubland**
/cc @vadimol @Slash32

### CODE REVIEW

**Comment by kaa-leeroy** [comment]
Can one of the admins verify this patch?

File: notificationdemo/source/c/src/kaa_demo.c [hunk 1]
```diff
@@ -24,21 +24,26 @@
...
-
 static kaa_client_t *kaa_client = NULL;
 
+enum color_t {RED, YELLOW, GREEN};
 
+static const char *enum_to_str(enum color_t code_color)
+{
+    switch(code_color) {
+        case RED:
+            return "CodeRed";
```
Comment (Slash32) [inline notificationdemo/source/c/src/kaa_demo.c:38]: Please add break

File: notificationdemo/source/c/src/kaa_demo.c [hunk 2]
```diff
@@ -24,21 +24,26 @@
...
+        case YELLOW:
+            return "CodeYellow";
```
Comment (Slash32) [inline notificationdemo/source/c/src/kaa_demo.c:41]: Please add break

File: notificationdemo/source/c/src/kaa_demo.c [hunk 3]
```diff
@@ -24,21 +24,26 @@
...
+        case GREEN:
+            return "CodeGreen";
```
Comment (Slash32) [inline notificationdemo/source/c/src/kaa_demo.c:44]: Please add break

File: notificationdemo/source/c/src/kaa_demo.c [hunk 4]
```diff
@@ -24,21 +24,26 @@
...
+        default:
```
Comment (Slash32) [inline notificationdemo/source/c/src/kaa_demo.c:46]: There should be no default case for pre-defined types (especially enums).

File: notificationdemo/source/c/src/kaa_demo.c [hunk 5]
```diff
@@ -24,21 +24,26 @@
...
-        return (error); \
-    }
-
-
-
 static kaa_client_t *kaa_client = NULL;
 
+enum color_t {RED, YELLOW, GREEN};
 
+static const char *enum_to_str(enum color_t code_color)
```
Comment (Slash32) [inline notificationdemo/source/c/src/kaa_demo.c]: Please rename the function to something like color_to_str.

File: notificationdemo/source/c/src/kaa_demo.c [hunk 6]
```diff
@@ -110,7 +115,10 @@ int main(/*int argc, char *argv[]*/)
      * Initialize Kaa client.
      */
     kaa_error_t error_code = kaa_client_create(&kaa_client, NULL);
-    KAA_DEMO_RETURN_IF_ERROR(error_code, "Failed create Kaa client");
+    if (error_code) {
+        printf("Failed create Kaa client %d\n", error_code);
```
Comment (Slash32) [inline notificationdemo/source/c/src/kaa_demo.c]: Please re-write the error message to common English variant.

File: notificationdemo/source/c/src/kaa_demo.c (hunk 4 above)
Comment (Dubland) [inline notificationdemo/source/c/src/kaa_demo.c]: It was discussed. There should be default case.

**Comment by Slash32** [comment]
LGTM

File: notificationdemo/source/c/src/kaa_demo.c [hunk 7]
```diff
@@ -24,21 +24,26 @@
...
-    if ((error)) { \
-        printf(message ", error code %d\n", (error)); \
-        return (error); \
-    }
-
-
-
 static kaa_client_t *kaa_client = NULL;
 
+enum color_t {RED, YELLOW, GREEN};
```
Comment (rasendubi) [inline notificationdemo/source/c/src/kaa_demo.c]: Don't use `_t` suffix -- it's reserved

File: notificationdemo/source/c/src/kaa_demo.c [hunk 8]
```diff
@@ -24,21 +24,28 @@
...
 
+static const char *color_to_str(enum color_t code_color)
+{
+    switch(code_color) {
+        case RED:
+            return "CodeRed";
+            break;
+        case YELLOW:
+            return "CodeYellow";
```
Comment (rasendubi) [inline notificationdemo/source/c/src/kaa_demo.c:41]: Still no break

**Comment by Dubland** [comment]
Fixed

File: notificationdemo/source/c/src/kaa_demo.c [hunk 9]
```diff
@@ -24,21 +24,29 @@
...
+            break;
+        case GREEN:
+            return "CodeGreen";
+            break;
+        default:
+            return "Incorrect value";
```
Comment (rasendubi) [inline notificationdemo/source/c/src/kaa_demo.c:47]: add break

**Comment by rasendubi** [comment]
LGTM

**Comment by vadimol** [comment]
+

**Pull request merged**